source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -572,6 +584,9 @@ void Filter::onGenericPoolReady(StreamInfo::StreamInfo* info,
   generic_conn_pool_.reset();
   read_callbacks_->upstreamHost(host);
   StreamInfo::UpstreamInfo& upstream_info = *getStreamInfo().upstreamInfo();
+  upstream_info.upstreamTiming().recordConnectionPoolCallbackLatency(
```

Comment:
Should we record the time when `onGenericPoolFailure`?

Reply:
The filter might try numerous times to connect to an upstream -- if it totally fails it will call `onInitFailure`.
As such we're recording the total latency of first attempt to when succeeds or when totally fails.

Reply:
I see now, thanks